pyaedt/modules/report_templates.py

```diff
@@ -1,8 +1,148 @@
+from pyaedt.generic.constants import LineStyle
+from pyaedt.generic.constants import SymbolStyle
+from pyaedt.generic.constants import TraceType
 from pyaedt.generic.general_methods import generate_unique_name
 from pyaedt.generic.general_methods import pyaedt_function_handler
 from pyaedt.modeler.GeometryOperators import GeometryOperators
 
 
+class LimitLine(object):
+    """Line Limit Management Class."""
+
+    def __init__(self, report_setup, trace_name):
+        self._oreport_setup = report_setup
+        self.line_name = trace_name
+        self.LINESTYLE = LineStyle()
+
+    @pyaedt_function_handler()
+    def _change_property(self, props_value):
+        self._oreport_setup.ChangeProperty(
+            ["NAME:AllTabs", ["NAME:Limit Line", ["NAME:PropServers", self.line_name], props_value]]
+        )
+        return True
+
+    @pyaedt_function_handler()
+    def set_line_properties(
+        self, style=None, width=None, hatch_above=None, violation_emphasis=None, hatch_pixels=None, color=None
+    ):
+        """Set Trace Properties.
+
+        Parameters
+        ----------
+        style : str or
+            Limit Line Style. LINESTYLE PROPERTY can be used
+        width : int
+            Limit Line width.
+        hatch_above : bool
+            Either if Hatch above the limit line or not.
+        violation_emphasis : bool
+            Either if to add violation emphasis or not.
+        hatch_pixels : int
+            Hatch pixel numbers.
+        color : tuple or list
+            Trace Color. a Tuple of (R,G,B) with int of [0,255] has to be provided.
```

Comment:
```suggestion
        style : str, optional
            Style for the limit line. The default is ``None``. You can also use the ``linestyle`` property.
        width : int, optional
            Width of the limit line. The default is ``None``.
        hatch_above : bool
           Whether the hatch is above the limit line. The default is ``None``.
        violation_emphasis : bool
            Whether to add violation emphasis. The default is ``None``.
        hatch_pixels : int
            Number of pixels for the hatch. The default is ``None``.
        color : tuple or list
            Trace color. A tuple of (R,G,B) with an integer list, such as [0,255], has to be provided.
```

Reply:
The above edits reflect what's shown in the function signature. However, what's there looks like it isn't final as the defaults don't seem to make sense. For example, defaults for booleans are generally "True" or "False". Also, when a default is "None," often times in needs an explanation of the behavior that occurs. For ``linewidth`` for example, "The default is ``None,`` in which case the line width is 1 point." In most cases here, the behavior is not obvious if "None`` is used.

Reply:
Hello @PipKat that's generally valid. Anyway I want to give the user the option to not "touch" that settings in AEDT by leaving to None the function parameters. If the user doesn't enter anything then the value in AEDT will not be touched. This is because at the moment there is no way to get actual status of the parameter in AEDT and if I force a default I would override user settings